read-the-docs/2/conf.py

```diff
@@ -19,26 +19,49 @@ def parse_branch_version():
       return os.environ['BRANCH_VERSION']
     return 'master'
 
-def parse_version_from_pom():
-
+def eval_version():
     # 1. get version from environment variable
     import os
     if "VERSION" in os.environ:
       return os.environ['VERSION']
 
     # 2. get version from pom.xml <VERSION>
+    pomVersion = parse_version_from_pom()
+    if pomVersion is not None:
+      release = pomVersion.rsplit('.',1)[0]
+      return release
+
+    # 3. fallback to dev version
+    return 'dev'
+
+def eval_version_patch():
+    # 1. get version from environment variable
+    import os
+    if "VERSION_PATCH" in os.environ:
+      return os.environ['VERSION_PATCH']
+
+    # 2. get version from pom.xml <VERSION>
+    pomVersion = parse_version_from_pom()
+    if pomVersion is not None:
+      return pomVersion
+
+    return "0.0.0"
+
+def parse_version_from_pom():
+    import os
     pomFile = os.environ['BASEDIR'] + '/pom.xml'
     if os.path.isfile(pomFile):
       from xml.etree import ElementTree as et
       tree = et.ElementTree()
       tree.parse(pomFile)
       ns = {"mvn":"http://maven.apache.org/POM/4.0.0"}
-      element = tree.getroot().find('./mvn:version', ns)
-      if element is not None:
-        return element.text
-
-    # 3. fallback to dev version
-    return 'dev'
+      version = tree.getroot().find('./mvn:version', ns)
+      if version is not None:
+        return version.text.removesuffix('-SNAPSHOT')
+      parentVersion = tree.getroot().find('./mvn:parent/mvn:version', ns)
+      if parentVersion is not None:
+        return parentVersion.text.removesuffix('-SNAPSHOT')
+    return None
```

Comment:
imho here is the main reason why this pom-version determination seldom works: we did not use the parent/version which is there in most of our pom.xml files.